drum/pattern.go

```diff
@@ -11,11 +14,11 @@ type Pattern struct {
 }
 
 func (p *Pattern) String() string {
-	str := ""
-	str += fmt.Sprintf("Saved with HW Version: %s\n", p.hw)
-	str += fmt.Sprintf("Tempo: %v\n", p.tempo)
+	buf := bytes.Buffer{}
```

Comment:
That's exactly what I wanted to see, a slightly better version using a byte buffer would look like that:

```
headerTxt := `Saved with HW Version: %s\nTempo: %v\n`
buf := bytes.NewBufferString(fmt.Sprintf(headerTxt, p.hw, p.tempo))
for _, track := range p.tracks {
    buf.WriteString(track.String())
}
return buf.String()
}
```